### PR TITLE
Add type hinting to JFormRule::test(). Fix an E_STRICT error in the process.

### PR DESCRIPTION
**Pull request by realityking**

### CODE REVIEW

**Comment by realityking** [comment]
I changed the type hints to SimpleXMLElement because of #1141.

File: libraries/joomla/form/field.php [hunk 1]
```diff
@@ -298,10 +298,10 @@ public function setForm(JForm $form)
 	 *
 	 * @since   11.1
 	 */
-	public function setup($element, $value, $group = null)
+	public function setup(JXmlElement $element, $value, $group = null)
```
Comment (LouisLandry) [inline libraries/joomla/form/field.php]: Didn't you want these to be `SimpleXMLElement` ?

Comment (realityking) [inline libraries/joomla/form/field.php]: You're right. Fixed.

**Comment by chdemko** [comment]
Cannnot be merged

**Comment by realityking** [comment]
Updated.

**Comment by eddieajau** [comment]
1 cs error and one ut error.

**Comment by realityking** [comment]
Thanks. They should be fixed now.

**Pull request merged**